### PR TITLE
add puppy_methods file we worked on in coderpad

### PR DESCRIPTION
**Pull request by tereg**
@bontgoy My pair and I weren't able to get our instance methods to appear right after the initialize method.  Right now our code runs the initialize method 50 times and our instance methods 50 times after that.  How can we fix this?

### CODE REVIEW

File: ruby/puppy_methods.rb [hunk 1]
```diff
@@ -7,3 +7,97 @@ def fetch(toy)
...
+
+
+p robert.saute
+p robert.number_of_dishes(30)
+
+z = 1
+chef_instances = {}
+
+
+until z == 51
```
Comment (Bontgoy) [inline ruby/puppy_methods.rb]: get rid of z and make this until

```
50.times do
     # code
end
```

File: ruby/puppy_methods.rb [hunk 2]
```diff
@@ -7,3 +7,97 @@ def fetch(toy)
...
+  chef = "chef #{z}"
+  chef.to_sym
+  puts chef
+  chef_instances[chef] = Chef.new
```
Comment (Bontgoy) [inline ruby/puppy_methods.rb]: `chef_instances[chef].saute` would work on line 92 but tell me why

File: ruby/puppy_methods.rb (hunk 1 above)
Comment (Bontgoy) [inline ruby/puppy_methods.rb]: you can still use the num if you want that for the chef name key though

**Comment by tereg** [comment]
Thanks, Ty!
In 91, when we create a new chef, we are adding a key value pair to the hash.
In 92, since we’re still in the loop, we run the saute method on the key we just added.  
We could do the same with number of dishes in line 93.

**Comment by Bontgoy** [comment]
Great!

**Pull request merged**